src/components/alarm/AlarmController.cpp

```diff
@@ -27,30 +27,45 @@ using namespace std::chrono_literals;
 AlarmController::AlarmController(Controllers::DateTime& dateTimeController) : dateTimeController {dateTimeController} {
 }
 
-APP_TIMER_DEF(alarmAppTimer);
-
 namespace {
-  void SetOffAlarm(void* p_context) {
-    auto* controller = static_cast<Pinetime::Controllers::AlarmController*>(p_context);
-    if (controller != nullptr) {
-      controller->SetOffAlarmNow();
-    }
+  void SetOffAlarm(TimerHandle_t xTimer) {
+    auto controller = static_cast<Pinetime::Controllers::AlarmController*>(pvTimerGetTimerID(xTimer));
+    controller->SetOffAlarmNow();
   }
 }
 
 void AlarmController::Init(System::SystemTask* systemTask) {
-  app_timer_create(&alarmAppTimer, APP_TIMER_MODE_SINGLE_SHOT, SetOffAlarm);
   this->systemTask = systemTask;
+  alarmAppTimer = xTimerCreate("alarmAppTm", 1, pdFALSE, this, SetOffAlarm);
```

Comment:
I would call this timer and variable just alarmTimer. `app_timer` is just what the nRF timer is called.

Reply:
Agreed, brief names for timers are better now as they are in the class scope.